editor/inserter/menu.js

```diff
@@ -307,16 +300,54 @@ export class InserterMenu extends Component {
 		);
 	}
 
+	renderBlocks( blocks, separatorSlug ) {
+		const { instanceId } = this.props;
+		const labelledBy = separatorSlug === undefined ? null : `editor-inserter__separator-${ separatorSlug }-${ instanceId }`;
+		return (
+			<div
+				className="editor-inserter__category-blocks"
+				tabIndex="0"
+				aria-labelledby={ labelledBy }
+			>
+				{ blocks.map( ( block ) => this.getBlockItem( block ) ) }
+			</div>
+		);
+	}
+
+	renderCategory( category, blocks ) {
+		const { instanceId } = this.props;
+		return blocks && (
+			<div key={ category.slug }>
+				<div
+					className="editor-inserter__separator"
+					id={ `editor-inserter__separator-${ category.slug }-${ instanceId }` }
+					aria-hidden="true"
+				>
+					{ category.title }
+				</div>
+				{ this.renderBlocks( blocks, category.slug ) }
+			</div>
+		);
+	}
+
+	renderCategories( visibleBlocksByCategory ) {
+		return getCategories().map(
```

Comment:
One final thought: Is there not a bit of redundancy in calling `getCategories` when `visibleBlocksByCategory` already includes this information? In other words, could we write this function as:

```js
return mapValues( visibleBlocksByCategory, ( blocks, category ) => (
	this.renderCategory( category, blocks ) )
) );
```

https://lodash.com/docs/4.17.4#mapValues